# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/HelperThreads.h
# Commit: 31d3f2901db9
# Full Hash: 31d3f2901db927a29898c5208a2624b4225382c1
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2024-12-05 09:58:04
# Regressor Bug: 1933681
# File Overlap Count: 1
# Description:
#   Bug 1933681 - Part 1: Allocate InitialStencilAndDelazifications in CompileGlobalScriptToStencilAndMaybeInstantiate. r=nbp
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D230387
# ==============================================================================

diff -r 2694b90c0deb -r 31d3f2901db9 js/src/vm/HelperThreads.h
--- a/js/src/vm/HelperThreads.h	Thu Dec 05 02:39:42 2024 +0200
+++ b/js/src/vm/HelperThreads.h	Wed Dec 04 23:44:33 2024 +0000
@@ -42,7 +42,7 @@
 class SourceCompressionTask;
 
 namespace frontend {
-struct CompilationStencil;
+struct InitialStencilAndDelazifications;
 }
 
 namespace gc {
@@ -246,9 +246,9 @@
 
 // Start off-thread delazification task, to race the delazification of inner
 // functions.
-void StartOffThreadDelazification(JSContext* maybeCx,
-                                  const JS::ReadOnlyCompileOptions& options,
-                                  const frontend::CompilationStencil& stencil);
+void StartOffThreadDelazification(
+    JSContext* maybeCx, const JS::ReadOnlyCompileOptions& options,
+    frontend::InitialStencilAndDelazifications* stencils);
 
 // Drain the task queues and wait for all helper threads to finish running.
 //
